org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/bundle/BundleArgumentsProvider.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.osgi.test.junit5.bundle;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.support.AnnotationConsumer;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+
+public class BundleArgumentsProvider implements ArgumentsProvider, AnnotationConsumer<BundleSource> {
+
+	private BundleSource bundleSource;
+
+	@Override
+	public Stream<? extends Arguments> provideArguments(ExtensionContext context) throws Exception {
+
+		BundleContext bundleContext = FrameworkUtil.getBundle(context.getClass())
```

Comment:
Similar to my comments below about services, this should more properly use the BundleContextExtension to get the bundle context.